Add EstablishmentModal component

jfplataroti opened the pull request:
Added `EstablishmentModal` component used in `DeliveryTabs` used for mobile screens.

Not a lot of changes, it follows the same behaviour than the `AddressModal` component. Also, it uses the `useEstablishment` hook used in `EstablishmentField` to load the establishment options.

Modal with geoLocation enabled:

<img width="472" alt="establishment picker" src="https://user-images.githubusercontent.com/1827213/107569763-ac11ce00-6bb6-11eb-9e50-280d927b8497.png">


Modal with geoLocation disabled:

![establishment-picker-modal-geo-disabled](https://user-images.githubusercontent.com/1827213/107569783-b2a04580-6bb6-11eb-8ccf-8510b11f6d2d.png)

djoyal commented on packages/web/components/deliveryTabs.tsx:
```diff
@@ -9,22 +9,30 @@ import ClockIcon from '../public/images/clock.svg';
+  const [showEstablishmentModal, setShowEstablishmentModal] = useState(false);
 
-  const onAddressFieldSelected = () => setShowDeliveryModal(true);
+  const openAddressModal = () => setShowDeliveryModal(true);
```
👍

djoyal commented on packages/web/components/establishmentModal/establishmentModal.tsx:
```diff
@@ -0,0 +1,44 @@
+  return (
+    <Layer onClickOutside={onClosed}>
+      <Box pad="medium" margin={{ top: 'small' }} width="large">
+        <Button fill={false} plain={true}>
```
You can actually condense this a bit more by embedding the icon directly in the button's props (E.G. `<Button icon={<Close />} />` :)

jfplataroti replied:
nice! I'll update it.

djoyal commented on packages/web/components/establishmentModal/suggestionListItem.tsx:
```diff
@@ -0,0 +1,61 @@
+  onSelected?: (establishment: Establishment) => void;
+}
+
+const formateDistance = (distance: number) =>
```
Extra `e` in format

djoyal commented on packages/web/components/establishmentModal/suggestionListItem.tsx:
```diff
@@ -0,0 +1,61 @@
+    : {};
+  const buttonStyle = { borderRadius: '12px', ...border };
+
+  const onClickHander = () => onSelected && onSelected(item);
```
`onClickHandler` doesn't tell me much about what this function does, could you use a more descriptive name please?

jfplataroti replied:
yes, sorry it doesn't follow the naming convention agreed in the previous PR.

jfplataroti replied:
updated :)

djoyal reviewed (approved):
The component looks great! There's a few small comments but nothing blocking. Nice work!

jfplataroti commented:
Hey @djoyal , I added the suggestions (the naming convention for the event handlers and simplifying the button using the icon attribute).

I added another commit to remove some code duplication when the address and distance were formatted for the establishment suggestions.

djoyal reviewed (approved):
Thanks for cleaning up the naming. I also really like moving the `formatAddress` function to the `lib/helpers`! 🙌
